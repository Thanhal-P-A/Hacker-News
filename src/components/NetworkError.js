import React from 'react';
import {Text, StyleSheet, View} from 'react-native';
import Color from '../common/Color';

export default function NetworkError() {
  return (
    <View style={styles.container}>
      <Text style={styles.label}>Internet not reachable</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    padding: 10,
    width: '100%',
    alignItems: 'center',
    backgroundColor: Color.Red,
  },
  label: {
    color: Color.White,
    fontSize: 16,
  },
});
