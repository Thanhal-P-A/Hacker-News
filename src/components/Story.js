import React from 'react';
import {View, Text, StyleSheet, TouchableOpacity, Linking} from 'react-native';
import moment from 'moment';

import {isEmpty} from '../common';
import Color from '../common/Color';

const Story = (item, index, onNavigate) => {
  const {by, descendants, kids, score, time, title, url} = item;
  const haveComments = !isEmpty(descendants) && descendants > 0;
  const canOpenURL = !isEmpty(url) && Linking.canOpenURL(url);
  return (
    <View style={styles.itemContainer}>
      <Text style={styles.textGrey}>
        {isEmpty(index) ? ' ▲ ' : `${index + 1}. ▲ `}
      </Text>
      <View style={{width: '90%'}}>
        <TouchableOpacity
          disabled={!canOpenURL}
          onPress={() => Linking.openURL(url)}>
          <Text style={styles.textBlack}>
            {title}
            <Text style={styles.textDetails}>{` (${
              url?.split('/')[2]
            }) `}</Text>
          </Text>
        </TouchableOpacity>
        <View style={{flexDirection: 'row'}}>
          <Text style={styles.textDetails}>
            {`${score} points by ${by} ${moment(time * 1000).fromNow()}`}
          </Text>
          {haveComments && (
            <TouchableOpacity
              disabled={isEmpty(onNavigate)}
              onPress={() => onNavigate(kids, item)}>
              <Text style={styles.textDetails}>
                {' | ' + descendants + ' Comments'}
              </Text>
            </TouchableOpacity>
          )}
        </View>
      </View>
    </View>
  );
};

export default Story;

const styles = StyleSheet.create({
  textBlack: {
    color: Color.Black,
  },
  itemContainer: {
    flexDirection: 'row',
    padding: 5,
    backgroundColor: Color.White,
  },
  textGrey: {
    color: Color.Grey,
  },
  textDetails: {
    color: Color.Grey,
    fontSize: 12,
  },
});
