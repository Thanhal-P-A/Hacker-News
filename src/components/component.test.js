import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import NetworkError from './NetworkError';
import Story from './Story';

it('NetworkError component renders correctly', () => {
  const NetworkErrorComponent = renderer.create(<NetworkError />).toJSON();
  expect(NetworkErrorComponent).toMatchSnapshot();
});

it('Story component renders correctly', () => {
  const StoryComponent = renderer.create(<Story />).toJSON();
  expect(StoryComponent).toMatchSnapshot();
});
