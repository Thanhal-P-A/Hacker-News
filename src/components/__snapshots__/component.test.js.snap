// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkError component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EB4234",
      "padding": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 16,
      }
    }
  >
    Internet not reachable
  </Text>
</View>
`;

exports[`Story component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexDirection": "row",
      "padding": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#707070",
      }
    }
  >
     â–² 
  </Text>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#707070",
              "fontSize": 12,
            }
          }
        >
           (undefined) 
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#707070",
            "fontSize": 12,
          }
        }
      >
        undefined points by undefined Invalid date
      </Text>
    </View>
  </View>
</View>
`;
